core/utils/exact_match_table.h

```diff
@@ -0,0 +1,599 @@
+// Copyright (c) 2014-2016, The Regents of the University of California.
+// Copyright (c) 2016-2017, Nefeli Networks, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// * Neither the names of the copyright holders nor the names of their
+// contributors may be used to endorse or promote products derived from this
+// software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef BESS_UTILS_EXACT_MATCH_TABLE_H_
+#define BESS_UTILS_EXACT_MATCH_TABLE_H_
+
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include <rte_config.h>
+#include <rte_hash_crc.h>
+
+#include "../message.h"
+#include "../metadata.h"
+#include "../module.h"
+#include "../packet.h"
+#include "bits.h"
+#include "cuckoo_map.h"
+#include "endian.h"
+#include "format.h"
+
+#define MAX_FIELDS 16
```

Comment:
move values related code out of `exact_match_table.h`. We can add a utils file if there is something to be shared by the exactmatch and wildcard values extension and include it.

Reply:
I did not add any extra macros to this file. All existing macros are part of the original `exact_match_table.h` file. Moving these would create a bigger diff fingerprint.

Reply:
MAX_FIELDS thought this was 8 [originally](https://github.com/NetSys/bess/blob/master/core/utils/exact_match_table.h#L50)?

Reply:
also what i meant was for this file to disappear from the diff if thats possible

Reply:
I have moved the code to exact_match module. MAX_FIELDS probably should be increased (left that macro to 16) since we are dealing with a lot of attributes in the current pipeline and we are testing this threshold.

Reply:
Any insight into effects or things to worry about growing beyond cacheline?

Reply:
Moved back MAX_FIELDS to 8.